source/extensions/filters/http/transformation/inja_transformer.cc

```diff
@@ -126,6 +126,15 @@ TransformerInstance::TransformerInstance(
   env_.add_callback("clusterMetadata", 1, [this](Arguments &args) {
     return cluster_metadata_callback(args);
   });
+  env_.add_callback("base64Encode", 1, [this](Arguments &args) {
+    return base64_encode_callback(args); 
+  });
+  env_.add_callback("base64Decode", 1, [this](Arguments &args) {
+    return base64_decode_callback(args); 
+  });
+  env_.add_callback("substring", 3, [this](Arguments &args) {
```

Comment:
curious if we can make the third arg here optional, which seems like a better UX to me

Reply:
I'll look into it -- I think we might get a render error if we don't get exactly 3 arguments, but It would be nice to have if that doesn't happen

Reply:
Did this here: https://github.com/solo-io/envoy-gloo/pull/184/commits/ef3a4251bf97736297be9bec0bdff14b9d5a4ed2

We just had to register two `substring` callbacks, one with two arguments and one with three